test/campaign-url-builder/problematic.js

```diff
@@ -0,0 +1,78 @@
+// Copyright 2016 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {expect} from 'chai';
+import ReactTestUtils from 'react-dom/test-utils';
+import renderProblematic from
+  '../../src/javascript/campaign-url-builder/components/problematic.js';
+
+// Given a basic domain, test the domain for all combinations of
+// leading protocol, subdomain, path, query, and fragment.
+const thoroughlyTestDomain = (domain, test) =>
+  ['', 'http://', 'https://'].forEach(proto =>
+    ['', 'www.', 'play.google.com.'].forEach(subdomain =>
+      ['', '/', '/object/path', '/trailing/slash/'].forEach(path =>
+        ['', '?utm_content=test'].forEach(query =>
+          ['', '#fragment'].forEach(fragment =>
+            test(`${proto}${subdomain}${domain}${path}${query}${fragment}`)
+          )
+        )
+      )
+    )
+  );
+
+describe('campaign-url-builder', () => {
+  describe('renderProblematic', () => {
+    it('does not flag ordinary URLs as problematic', () => {
+      ['example.com', 'mail.google.com']
+      .forEach(domain => {
+        thoroughlyTestDomain(domain, url => {
+          const renderResult = renderProblematic(url);
+          expect(renderResult)
+            .to.be.an('object')
+            .that.deep.equals({element: null, eventLabel: null});
+        });
+      });
+    });
+
+    it('returns a React element and analytics alert for problematic URLs',
+      () => {
```

Comment:
If you're going to line-break here, this should be indented 4 spaces.

Reply:
Do multi-line function arguments not count as a block construct? I figured that that parenthesis, and the fact that it isn't really an expression continutation, made the block rule apply

Reply:
I think the only things that count as block constructs are object and array literals.